torchbox/torchbox.com
# Markdown block and grouping of streamfield blocks

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1362062112)

### Description of Changes Made

- Adds styling for the markdown block
- Customise the options for the markdown block so that only the code block is available - this requires some JS just for the admin area, and consequently a new entry point for the js webpack.
- Ensures the 'overflow:scroll' for the markdown block is focusable
- Groups the various streamfield blocks to make life easier for the editor
- Note that I have not styled headings etc for markdown - I've added help text to use this for code only.

### How to Test

- Edit a standard page and add a markdown block. Try adding some code, e.g.

```html
<html>
<a href="foo">bar</a>
</html>
```

- Ensure the code has syntax highlighting as expected.
- Edit a service page and the home page - test adding a new block on these and check the groupings make sense.

### Screenshots

<details>
  <summary>Expand to see more</summary>

With scrolling

![Screenshot 2024-03-18 at 10 30 07](https://github.com/torchbox/torchbox.com/assets/771869/fd5cf7e6-7606-43d1-a21d-0c7278f6917a)

Syntax highlighting
![Screenshot 2024-03-18 at 10 29 38](https://github.com/torchbox/torchbox.com/assets/771869/dac2bc46-2d75-4f30-a3af-94fcaf1afb69)

New groupings

![Screenshot 2024-03-18 at 10 30 58](https://github.com/torchbox/torchbox.com/assets/771869/0cc94292-53ee-43bb-a7e7-34a086c9b403)

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
